### ISSUE TITLE
Verify pinout of TO* footprints

### ISSUE THREAD

**Issue by odurc** [KiCad/TO_SOT_Packages_THT.pretty#6]
According #3 the following footprints may be wrong: TO-12, 18, 43, 72, 3.

**Comment by evanshultz** [comment]
I was able to find some of these packages at:
- http://www.ti.com/lit/an/snoa033/snoa033.pdf
- http://www.interfacebus.com/semiconductor-transistor-packages.html
- http://www.onsemi.com/pub/Collateral/MJ11012-D.PDF
- http://www.st.com/content/ccc/resource/technical/document/datasheet/1e/a2/bd/86/2f/6f/46/f0/CD00002665.pdf/files/CD00002665.pdf/jcr:content/translations/en.CD00002665.pdf
- https://www.microsemi.com/document-portal/doc_view/132304-lds-0307

TO-18 is correct:
![image](https://user-images.githubusercontent.com/1936989/28949496-f944c4e8-7871-11e7-9f2a-c04c979cd7f8.png)
![image](https://user-images.githubusercontent.com/1936989/28949363-2b54e626-7871-11e7-8d0b-6f93cd108c83.png)

TO-72 is correct:
![image](https://user-images.githubusercontent.com/1936989/28949478-e5137fc8-7871-11e7-9f84-fd87f8a6ebc8.png)
![image](https://user-images.githubusercontent.com/1936989/28949388-5630a2ea-7871-11e7-85ec-35a4288c43c5.png)

TO-3 seems to vary:
![image](https://user-images.githubusercontent.com/1936989/28949467-da80fde2-7871-11e7-9e5e-e971db842267.png)
![image](https://user-images.githubusercontent.com/1936989/28950056-cf2e6fac-7875-11e7-905e-aa836fc53006.png)
![image](https://user-images.githubusercontent.com/1936989/28950027-9597d210-7875-11e7-9599-f6d2814b1ae1.png)
![image](https://user-images.githubusercontent.com/1936989/28950047-b82b52e8-7875-11e7-9e2c-48eb2bbcf171.png)
![image](https://user-images.githubusercontent.com/1936989/28949460-cfe39cf0-7871-11e7-8462-5fd866cd5aaf.png)

So I believe the KiCad library has been updated since this issue was posted. I'll keep looking for package drawings of the other 2 to confirm them.

**Comment by jkriege2** [comment]
@evanshultz Will you post a PR for TO-3?

**Comment by evanshultz** [comment]
@jkriege2 Of course! But I'm stuck on the name. Having pins 1 and 2 swapped don't form any kind of natural convention. `Pin1_Top` and `Pin1_Bottom` are too lame. Suffixing `_Reverse` is the only thing I can think of that's universal (and used elsewhere in our library), but that's not very intuitive for the CvPcb/Pcbnew user. Do you have a suggestion?

**Comment by jkriege2** [comment]
I had a quick glance: The package drawings above for TO-3 are BOTTOM views, if you rotate them correctly to the TOP view that we see in KiCAD, the numbering in the screenshots matches + all symbols that use the TO-3 should be OK (just checked them again against the current footprint and the respective DSs):

- regul.lib:LM317_TO3: Here the current symbol is OK (TI datasheet gives pin-numbers as they are currently: http://www.ti.com/lit/ds/symlink/lm117.pdf) ![2017-08-04 19_04_10-lm117 pdf - adobe acrobat reader dc](https://user-images.githubusercontent.com/11967523/28978803-c122d196-7947-11e7-94ca-2777a46e8a1d.png)
- regul.lib:LM337_TO3 ... same as above (https://www.elektronik-kompendium.de/public/schaerer/FILES/lm337.pdf)
- transistors.lib:2N3055 pin-numbering and connections match current footprint and DS: http://www.onsemi.com/pub/Collateral/2N3055-D.PDF
- transistors.lib:MJ2955 pin-numbering and connections match current footprint and DS: http://www.onsemi.com/pub/Collateral/2N3055-D.PDF


So, in conclusion I would say the TO-3 is correct ... or did I overlook something? Can we close this then?

JAN

**Comment by evanshultz** [comment]
I (disturbingly) can't find a TO-12 package drawing online, but our footprint has pin 1 as the first pin CCW from the tab, which is consistent with all the other TO footprints in the lib:
![image](https://user-images.githubusercontent.com/1936989/28978878-a3740b92-78fc-11e7-91c3-b2a5b51a45dc.png)

I don't find TO-43 footprints at https://github.com/KiCad/TO_SOT_Packages_THT.pretty, but perhaps that was a typo and should be TO-46? What I do see is that our TO-46 footprint matches the parts I see:
http://www.mouser.com/ds/2/187/HFE4070-500_serie-191281.pdf
![image](https://user-images.githubusercontent.com/1936989/28979035-18635ea8-78fd-11e7-971a-e2c762f90931.png)
http://cds.linear.com/docs/en/controlled-drawings/RH129_05-08-5027_SPEC_REV.H.pdf
![image](https://user-images.githubusercontent.com/1936989/28979093-4bad26e0-78fd-11e7-98f1-51016da39ba3.png)
http://www.ti.com/lit/ds/symlink/lm34.pdf
![image](https://user-images.githubusercontent.com/1936989/28979130-6a033e68-78fd-11e7-8854-df51bcc4a12c.png)
![image](https://user-images.githubusercontent.com/1936989/28979004-fc44ef2a-78fc-11e7-8738-cd73f5514b57.png)

So with regards to everything but TO-3, I think it's fine. I would like to find a TO-12 part datasheet.

@jkriege2 
In response to you, Jan, I do have one outstanding question about TO-3. All of your work above looks great and seems to match. But I posted a [Microsemi datasheet](https://www.microsemi.com/document-portal/doc_view/132304-lds-0307) above which was the only outlier. It's the last package drawing I posted before the KiCad TO-3 footprint. It also appears to be a bottom view but the pin numbers and pinout are reversed.

Let me separate this out a bit:
- In the KiCad TO-3 screenshot above, pin 1 is at the top.
- [2N3055 from On Semi](http://www.onsemi.com/pub/Collateral/2N3055-D.PDF) also has pin 1 at the top and is the base connection. This seems to be common.
- However, the Microsemi part calls the top pin pin2 and that's the emitter.

So if I read this all right, there is a mismatch. Sorry for not going into more details about exactly where I saw the conflict earlier.

**Comment by jkriege2** [comment]
@evanshultz yes, now I see ... sorry ;-) ... mmhhh ... looking a round a bit I also found an IRF part that has this pinout: http://www.irf.com/product-info/datasheets/data/irfag50.pdf ... they call this package TO-204AA ... but then that name is also in this 2N3055 datasheet http://www.onsemi.com/pub/Collateral/2N3055-D.PDF ... i.e. a mess ;-)

Currently I'm tempted to say we stick to one symbol and let the user ensure that he get's then pin-mapping right, because creating two that only differ in the pin ordering might raise even more uncertainty? the work to check the pin-ordering against the DS during review is the same with one or two footprints ...

On the other hand, having two footprint options (e.g. `TO-3_Style2`and `TO-3_Style1`) might raise the awareness taht there are different interretations in future ...

Don't really know how to proceed on this ...

JAN

**Comment by evanshultz** [comment]
@jkriege2 
Great find! To me, this adds even more reason to make an alternate footprint. Otherwise, since the symbol pinout is fixed, we have a situation where a user could select a part and put a symbol on their schematic, but there's no correct footprint. They might quit KiCad or use the wrong footprint, which would be a bad.

If two footprints show up based on the FPfilter, then at least the user will need to think and find the right one. That seems more safe and just better, to me.

But that doesn't resolve how to name it. I mentioned adding the suffix `_Reverse` above, but I'm not in love with it. Is there something better you can think of?

**Comment by jkriege2** [comment]
What about `TO-3_SwappedPins12`? Also not really nice, but WYRIWYG (what you read is what you get ;-)

JAN

**Comment by evanshultz** [comment]
I don't like that footprint name. Since we find the existing footprint more common, but not a standard or obvious (let's say you're most familiar with Microsemi parts), what is it swapped from? And then neither footprint has information to help the user until they look at the footprint. (Neither does adding `Reverse`, for that matter, but it's more generic.)

Any other footprint name ideas?

**Comment by bobc** [comment]
> However, the Microsemi part calls the top pin pin2 and that's the emitter.

I suspect the Microsemi datasheet is incorrect, the diagram on page 8 does not page 7!

**Comment by evanshultz** [comment]
@bobc I see what you mean. The table on page 7 does not match the schematic on page 8. They both use `T1` and `T2`, but the pins assignments are swapped between the pages. Probably in error.

With that, I'm closing this case and considering it resolved.
